### PR TITLE
#92 Sync slider controls.

### PR DESCRIPTION
**Pull request by zuev93**
Example: Manual Generator, Coal generator.
Removed SideScreenEvents.cs :) Idea does not really scale for more complex side screens and requires a lot of exceptions.

### CODE REVIEW

File: MultiplayerMod/Game/Mechanics/SliderEvents.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using MultiplayerMod.Core.Patch;
+using MultiplayerMod.Multiplayer.Objects;
+
+namespace MultiplayerMod.Game.Mechanics;
+
+public static class SliderEvents {
+
+    public static event Action<MultiplayerReference, float, int> SetSliderValue;
+
+    [HarmonyPatch]
```
Comment (polycone) [inline MultiplayerMod/Game/Mechanics/SliderEvents.cs:15]: You can use it on outer class without any problems.

File: MultiplayerMod/Game/Mechanics/SliderEvents.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using System.Reflection;
+using HarmonyLib;
+using MultiplayerMod.Core.Patch;
+using MultiplayerMod.Multiplayer.Objects;
+
+namespace MultiplayerMod.Game.Mechanics;
+
+public static class SliderEvents {
+
+    public static event Action<MultiplayerReference, float, int> SetSliderValue;
```
Comment (polycone) [inline MultiplayerMod/Game/Mechanics/SliderEvents.cs:13]: Yeah, that's what I told you about, unknown "float, int"🙂

**Review by polycone — APPROVED** [review]

File: MultiplayerMod/MultiplayerMod.csproj [hunk 3]
```diff
@@ -29,4 +29,7 @@
         <Reference Include="UnityEngine.IMGUIModule" HintPath="$(ManagedPath)\UnityEngine.IMGUIModule.dll" Private="false" />
         <Reference Include="ImGui.NET" HintPath="$(ManagedPath)\ImGui.NET.dll" Private="false" />
     </ItemGroup>
+    <ItemGroup>
+      <Folder Include="Multiplayer\Commands\Gameplay\SliderControl" />
```
[the remark this answers is not in the document]
Comment (zuev93) [reply MultiplayerMod/MultiplayerMod.csproj:33]: Done

File: MultiplayerMod/Game/Mechanics/SliderEvents.cs (hunk 1 above)
Comment (zuev93) [reply MultiplayerMod/Game/Mechanics/SliderEvents.cs:15]: Done

File: MultiplayerMod/Game/Mechanics/SliderEvents.cs (hunk 2 above)
Comment (zuev93) [reply MultiplayerMod/Game/Mechanics/SliderEvents.cs]: Done

**Pull request merged**